### PR TITLE
Refactor main.js

### PR DESCRIPTION
**Pull request by ahmedalsudani**
@fjvallarino @tejohnso 

This PR is mostly going to receive additions and tweaks from this point on, but I don't plan on any major changes. Let me know if you find any room for improvement.

To get it to a ready state, it needs to include complete test coverage. Specifically:
- cover the `ready` handler
- try to test the `all-closed` handler

It will also undergo a line-by-line review before being merged to ensure no lines were erroneously deleted in the process of refactoring

### CODE REVIEW

**Comment by ahmedalsudani** [comment]
@fjvallarino @tejohnso ready for review.

Before merging:

- the commits will be squashed into one
- `main.js` and `main-controller.js` will be compared line for line to ensure no statements were lost

File: installer/main-controller.js [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+mainWindow;
+
+module.exports = {
+  // Yuck
+  init(imports) {
+    log.external("started", module.exports.isUnattended() ? "" : "attended");
+    var importKeys = ["app", "displaySettings", "ipc", "ui"];
+    importKeys.forEach((key)=>{
+      // Catches null and undefined
+      if (imports[key] == null) {
```
Comment (tejohnso) [inline installer/main-controller.js]: @ahmedalsudani if (!imports[key]) {}  ?

Comment (ahmedalsudani) [inline installer/main-controller.js]: @tejohnso that would work in this case, but the convention I'm used to is that `false` and `0` are considered valid values.

`!imports[key]` would cast `0` or `false` to `true`

**Comment by tejohnso** [comment]
@ahmedalsudani commit added for test timing issues.

**Comment by fjvallarino** [comment]
:+1:

**Comment by ahmedalsudani** [comment]
@fjvallarino @tejohnso the latest commit was a rebase+squash

**Comment by tejohnso** [comment]
:+1:

**Pull request merged**